src/ripple/ledger/impl/ApplyStateTable.cpp

```diff
@@ -0,0 +1,589 @@
+                        finals.emplace_back (obj);
+                }
+
+                if (!finals.empty ())
+                    meta.getAffectedNode (item.first).emplace_back (std::move(finals));
+            }
+            else if (type == &sfCreatedNode) // if created, thread to owner(s)
+            {
+                assert (curNode && !origNode);
+                threadOwners (to, meta, curNode, newMod, j);
+
+                if (curNode->isThreadedType ()) // always thread to self
+                    threadTx (meta, curNode, newMod);
+
+                STObject news (sfNewFields);
+                for (auto const& obj : *curNode)
+                {
+                    // save non-default values
+                    if (!obj.isDefault () &&
+                            obj.getFName().shouldMeta(
+                                SField::sMD_Create | SField::sMD_Always))
+                        news.emplace_back (obj);
+                }
+
+                if (!news.empty ())
+                    meta.getAffectedNode (item.first).emplace_back (std::move(news));
+            }
+            else
+            {
+                assert (false);
+            }
+        }
+
+        // add any new modified nodes to the modification set
+        for (auto& mod : newMod)
+            update (to, mod.second);
+
+        sMeta = std::make_shared<Serializer>();
+        meta.addRaw (*sMeta, ter, to.txCount());
+
+        // VFALCO For diagnostics do we want to show
+        //        metadata even when the base view is open?
+        JLOG(j.trace) <<
+            "metadata " << meta.getJson (0);
+    }
+    to.rawTxInsert(
+        tx.getTransactionID(),
+            sTx, sMeta);
+    apply(to);
+}
+
+//---
+
+bool
+ApplyStateTable::exists (ReadView const& base,
+    Keylet const& k) const
+{
+    auto const iter = items_.find(k.key);
+    if (iter == items_.end())
+        return base.exists(k);
+    auto const& item = iter->second;
+    auto const& sle = item.second;
+    switch (item.first)
+    {
+    case Action::erase:
+        return false;
+    case Action::cache:
+    case Action::insert:
+    case Action::modify:
+        break;
+    }
+    if (! k.check(*sle))
+        return false;
+    return true;
+}
+
+auto
+ApplyStateTable::succ (ReadView const& base,
+    key_type const& key, boost::optional<
+        key_type> const& last) const ->
+            boost::optional<key_type>
+{
+    boost::optional<key_type> next = key;
+    items_t::const_iterator iter;
+    // Find base successor that is
+    // not also deleted in our list
+    do
+    {
+        next = base.succ(*next, last);
+        if (! next)
+            break;
+        iter = items_.find(*next);
+    }
+    while (iter != items_.end() &&
+        iter->second.first == Action::erase);
+    // Find non-deleted successor in our list
+    for (iter = items_.upper_bound(key);
+        iter != items_.end (); ++iter)
```

Comment:
I just checked to make sure, and generally we indent at least to the `(` - in order not to be confused with the enclosed scope.  We don't have that many ifs over two lines, but all of them seem to do this.